.github/workflows/pull-db-tests.yml

```diff
@@ -86,7 +86,7 @@ jobs:
     runs-on: ubuntu-latest
     services:
       mysql:
-        image: mysql:5.7
+        image: mysql:8.1
```

Comment:
It should test with MySQL 8.0 but not 8.1

The same to pgsql/mssql

Reply:
Why not test with latest? New deployments are likely to use latest. I see no point testing non-latest if if only test one version.

Reply:
This one pipeline I guess we could move to 8.0 because 8.1 is tested below separately, but I don't see much of a point either.

Reply:
What if you used a MySQL 8.1 only syntax?

Reply:
By your notion, we would have to test every single supported db version, which given our CI resources is likely not possible.

Reply:
Maybe we should stick to MySQL LTS versions? According to this 8.0, 8.4, etc will be LTS:

![image](https://github.com/go-gitea/gitea/assets/115237/4143e8fc-8cfd-412e-a891-8a917df6b662)

Reply:
Ideally, MariaDB should have a separate test.

For others, databases have pretty good backward compatibility. If MySQL 8.0 passes, then MySQL 8.1 are almost sure to pass.

Reply:
> Maybe we should stick to MySQL LTS versions? According to this 8.0, 8.4, 9.7 will be LTS:

No, at the moment, only the minimal supported version is enough.

Reply:
I still don't understand the reasons. We can:

- Test latest, be most likely correct for new deployment.
- Test oldest, be correct for ancient deployments, but unknown for new ones.
- Test something in between, but by what rules?

I think it's easier to maintain if we just test latest, so that the maintainer updating this doesn't have to thing about obscure rules about which version to test.

Ideally we'd have a matrix build to easily add/remove versions. Currently it's a pain because of all the Makefile changes and such. If we had a matrix, I guess testing oldest and newest of each version would be ideal.

Reply:
> * Test latest, be most likely correct for new deployment, but old ones may break with incompatibly SQL syntax etc.

No, it would break the old database easily.

> * Test oldest, be correct for ancient deployments, but unknown for new ones.

This is the right approach. It won't be "unknown": see my comment: `databases have pretty good backward compatibility. If MySQL 8.0 passes, then MySQL 8.1 are almost sure to pass.`

> * Test something in between, but by what rules?

No, it's not worth to be considered.

Reply:
This PR doesn't change the behaviour of what we are doing with tests (w/ psql we're using latest, and w/ mqsl we were using 8.x).

I agree we should test both min/max db versions, but that can be done in a different PR.